powerspikegg/gateway/src/test/scala/gg/powerspike/MockGrpcServer.scala

```diff
@@ -0,0 +1,28 @@
+package gg.powerspike
+
+import io.grpc.ServerBuilder
+
+/**
+  * Grpc server which implements services necessary to test the gateway
+  */
+class MockGrpcServer {
```

Comment:
You should definitely not separate server setup logic from the test. The mock is defined in MockFetcher.scala, which is kinda legitimate (even if it's arguable since the mock does not implement important logic), but it does not make that much sense to separate the server setup logic.

IMHO you should use [fixtures](http://www.scalatest.org/user_guide/sharing_fixtures) to set up this server. You can avoid trying to make it generic among the tests.
Alternatively, you can put the server set up logic into a function in the mock but it sounds weird to me.